src/main/java/eu/europeana/metis/sandbox/service/workflow/MediaProcessingServiceImpl.java

```diff
@@ -76,17 +78,51 @@ public RecordInfo processMedia(Record record) {
     return new RecordInfo(Record.from(record, outputRdf), recordErrors);
   }
 
+  /**
+   * Processes the resource to get the main thumbnail
+   *
+   * @return True if the resource had a main thumbnail, therefore the method is successful; False
+   * otherwise.
+   */
+  private boolean processMainThumbnailResource(RdfDeserializer rdfDeserializer, byte[] inputRdf,
+      MediaExtractor extractor, Record record, List<RecordError> recordErrors) {
+
+    ResourceExtractionResult extractedThumbnail = null;
+    try {
+      RdfResourceEntry mainThumbnail = rdfDeserializer
+          .getMainThumbnailResourceForMediaExtraction(inputRdf);
+      extractedThumbnail = extractor.performMediaExtraction(mainThumbnail, false);
+    } catch (RdfDeserializationException | MediaExtractionException e) {
+      LOGGER.warn("Error while extracting media for record {}. ", record.getRecordId(), e);
+      recordErrors.add(new RecordError(new RecordProcessingException(record.getRecordId(), e)));
+    }
+
+    boolean hasThumbnail =
+        extractedThumbnail != null && !extractedThumbnail.getThumbnails().isEmpty();
```

Comment:
Here you should also call `rdfForEnrichment.enrichResource` (if `extractedThumbnail` is not null). In fact, the result is not just a thumbnail, but actually a media extraction that may or may not have a thumbnail (but if it does it is the main thumbnail).

Reply:
Done